File: BaekJoon/ITBigTech/ImsMiniGame/ImsMiniGame.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {

	reader := bufio.NewReader(os.Stdin)
	userInput, _ := reader.ReadString('\n')
	userInput = strings.TrimSpace(userInput)
	input := strings.Split(userInput, " ")

	N, _ := strconv.Atoi(input[0])
	game := input[1]

	names := make(map[string]struct{})

	for i := 0; i < N; i++ {
		name, _ := reader.ReadString('\n')
		name = strings.TrimSpace(name)
		names[name] = struct{}{}
	}

	if game == "Y" {
		fmt.Println(len(names))
	} else if game == "F" {
		fmt.Println(len(names) / 2)
	} else if game == "O" {
		fmt.Println(len(names) / 3)
	}

}

File: BaekJoon/ITBigTech/ImsMiniGame25757/ImsMiniGame.py
def main():
    N, games = input().split(" ")
    N = int(N)
    # s = set()
    # for _ in range(N):
    #   s.add(input())
    s = {input() for _ in range(N)}

    if games == 'Y':
        print(len(s))

    if games == "F":
        print(len(s)//2)

    if games == "O":
        print(len(s)//3)


if __name__ == "__main__":
    main()